crates/anemo/src/network/connection_manager.rs

```diff
@@ -227,13 +226,67 @@ impl ConnectionManager {
 
     fn handle_incoming(&mut self, connecting: Connecting) {
         trace!("received new incoming connection");
-        let connecting = connecting.map(|connecting_result| ConnectingOutput {
+
+        self.pending_connections.spawn(Self::handle_incoming_task(
+            connecting,
+            self.config.clone(),
+            self.active_peers.clone(),
+            self.known_peers.clone(),
+        ));
+    }
+
+    async fn handle_incoming_task(
+        connecting: Connecting,
+        config: Arc<Config>,
+        active_peers: ActivePeers,
+        known_peers: KnownPeers,
+    ) -> ConnectingOutput {
+        let fut = async {
+            let connection = connecting.await?;
+
+            // TODO close the connection explicitly with a reason once we have machine
+            // readable errors
```

Comment:
Maybe add explicit reference to issue #13? (or is this referring to some other machine readable errors?)